### PR TITLE
Fix IORedis Support

### PR DESCRIPTION
**Pull request by emeraldarcher**
The PR by @dearpeach to add support for IORedis which returns transaction results in a different format than node_redis was never resolved. This PR adds support for IORedis transactions while maintaining the existing support for node_redis. Ultimately I decided on a slightly different approach from @dearpeach. I have included a commented out test line for using IORedis as the client in the test file and have verified all tests pass for each client.

### CODE REVIEW

**Comment by emeraldarcher** [comment]
@facundoolano I've completed the changes to support IORedis' different transactional response format found here: https://github.com/luin/ioredis/wiki/Migrating-from-node_redis. Please review at your earliest convenience.

File: index.js [hunk 1]
```diff
@@ -11,6 +11,18 @@ function asPromise (fn) {
   });
 }
 
+function isArray (item) {
+  return Array.isArray(item);
+}
+
+function isIORedisFormattedTransactionResults (results) {
```
Comment (facundoolano) [inline index.js]: How about a single function

```
function convertToIORedis (results) {
    if Array.isArray(results[0]) {
        return results.map((result) => result[1]);
    }
    return results;
}
```

So below you can do 
```
return asPromise(multi.exec.bind(multi))
    .then(convertToIORedis)  
    .then((results) => {
```

Comment (facundoolano) [reply index.js]: That convertToIORedis could have a comment explaining why this convertion is necessary

**Comment by facundoolano** [comment]
@emeraldarcher this looks good. I left a suggestion.

**Comment by emeraldarcher** [comment]
@facundoolano I've updated per your suggestion. I left the `isArray` function because it is also used in the `values` function.

**Comment by emeraldarcher** [comment]
@facundoolano Thank you :)

**Pull request merged**